maven-plugin/src/main/java/org/openmrs/maven/plugins/Pull.java

```diff
@@ -0,0 +1,293 @@
+    private static final String CREATING_LOCAL_REPO_ERROR = "problem when initializing local repository";
+    private static final String GIT_COMMAND_ERROR = "problem with resolving git command";
+    private static final String CREATING_TEMP_REMOTE_ERROR = "problem with creating temporary remote to perform automatic update";
+
+    /**
+     * @parameter expression="${serverId}"
+     */
+    private String serverId;
+
+    private String failureReason;
+
+    public Pull() {}
+
+    public Pull(AbstractTask other) {super(other);}
+
+    public Pull(MavenProject project, MavenSession session, BuildPluginManager manager) {
+        super.mavenProject = project;
+        super.mavenSession = session;
+        super.pluginManager = manager;
+    }
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+        serverId = wizard.promptForExistingServerIdIfMissing(serverId);
+        Server server = Server.loadServer(serverId);
+
+        if(server.hasWatchedProjects()){
+            boolean pullingError = false;
+            Set<Project> watchedProjects = server.getWatchedProjects();
+            for(Project project: watchedProjects) {
+                wizard.showMessage(String.format(UPDATING_MODULE_MESSAGE, project.getArtifactId()));
+                if (pullLatestUpstream(project.getPath())) {
+                    wizard.showMessage(String.format(PULL_SUCCESS_MESSAGE, project.getArtifactId()));
+                } else {
+                    getLog().error(String.format(PULL_ERROR_MESSAGE, project.getArtifactId(), failureReason));
+                    getLog().error(UPDATE_MANUALLY_MESSAGE);
+                    pullingError = true;
+                }
+            }
+            if(pullingError){
+                throw new MojoExecutionException("Some modules could not be updated automatically, please see error messages above");
+            }
+        }else {
+            wizard.showMessage(String.format(NO_WATCHED_MODULES_MESSAGE, serverId));
+        }
+    }
+
+    /**
+     * Pull latest changes from upstream master
+     *
+     * @param path
+     * @return true if pull was successful
+     * @throws MojoExecutionException
+     */
+    private boolean pullLatestUpstream(String path) throws MojoExecutionException {
+        Repository localRepo;
+        Git git = null;
+        String previousBranch;
+        String newBranchFull;
+        String newBranch = "sdk-" + serverId;
+        try {
+            localRepo = new FileRepository(new File(path, ".git").getAbsolutePath());
+            git = new Git(localRepo);
+
+            previousBranch = localRepo.getBranch();
+
+            checkoutAndCreateNewBranch(git, newBranch);
+            newBranchFull = localRepo.getFullBranch();
+            RevCommit stash = git.stashCreate().call();
+            addRemoteTempRepo(git, path, newBranch);
```

Comment:
To be more specific, if the user didn't add remote to the upstream with the url from pom.xml JGit won't execute pull command